### ISSUE TITLE
Gleam stdlib fails to build on Windows

### ISSUE THREAD

**Issue by ShaneEverittM** [gleam-lang/gleam#2032]
I installed gleam 0.22.1 on Windows 11 using Chocolatey and ran:

```shell
gleam new hello_world
cd hello_world
gleam test
```
and saw the following output:
```shell
C:\Users\shane\dev\gleam\hello_world〉gleam run                                                                                         02/27/2023 06:43:54 PM
  Resolving versions
Downloading packages
 Downloaded 2 packages in 0.26s
  Compiling gleam_stdlib
error: Syntax error
    ┌─ build\packages\gleam_stdlib\src\gleam\dynamic.gleam:332:22
    │
332 │     use inner_result <- result.then(decode_result(value))
    │                      ^ This operator has no value on its right side.

Hint: Remove it or put a value after it.
```

**Comment by ShaneEverittM** [comment]
I switched to the pre-compiled release and get a similar error.
```shell
C:\Users\shane\dev\gleam\hello_world〉gleam run                                                                                         02/27/2023 09:08:37 PM
  Resolving versions
  Compiling gleam_stdlib
error: Syntax error
    ┌─ build\packages\gleam_stdlib\src\gleam\string.gleam:860:9
    │
860 │     let assert Ok(string) =
    │         ^^^^^^ I was not expecting this.

Expected one of:
A pattern
```

**Comment by inoas** [comment]
In `gleam.toml` we specify to auto upgrade stdlib. sadly 0.27 stdlib only runs with gleam 0.27 (currently rc1).

Please edit gleam.toml and change the version for the stdlib to `~> 0.26.0`, then potentially run `gleam update` and then it should work I hope.

For further help visit our discord community: https://discord.com/invite/Fm8Pwmy

Please note that gleam 0.22.1 is very outdated.

**Comment by inoas** [comment]
Also see https://discord.com/channels/768594524158427167/1079779667176853677

**Comment by lpil** [comment]
Hello! Thanks for trying Gleam.

The issue here is that 0.22 is rather old, we're currently on 0.27.0-rc1. If you upgrade the issue will go away.

**Comment by ShaneEverittM** [comment]
I know this issue is closed, but with gleam 0.26.2 and gleam stdlib 0.26 I still get an error with "let assert" as described by @inoas.

**Comment by ShaneEverittM** [comment]
Nevermind! Since I opened this it seems 0.27 has been released, which works!
